lib/onebox/engine/discourse_local_onebox.rb

```diff
@@ -51,7 +51,7 @@ def to_html
             topic = post.topic
             slug = Slug.for(topic.title)
 
-            excerpt = post.excerpt(SiteSetting.post_onebox_maxlength)
+            excerpt = post.excerpt(SiteSetting.post_onebox_maxlength, { keep_emoji_codes: true })
```

Comment:
This case is a little special because we're getting the cooked version and then passing it again to the markdown engine. Because emojis have `alt=':heart:'`, the markdown engine is coverting `:heart:` to another emoji image.